assets/src/edit-story/components/panels/document/publish/author.js

```diff
@@ -0,0 +1,128 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { __ } from '@web-stories-wp/i18n';
+import { useCallback, useEffect, useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { AdvancedDropDown, Row } from '../../../form';
+import { useStory } from '../../../../app/story';
+import { useAPI } from '../../../../app/api';
+import useInspector from '../../../inspector/useInspector';
+
+function Author() {
+  const {
+    actions: { getAuthors },
+  } = useAPI();
+  const {
+    state: { tab, users, isUsersLoading },
+    actions: { loadUsers },
+  } = useInspector();
+  const { isSaving, author, updateStory } = useStory(
+    ({
+      state: {
+        meta: { isSaving },
+        story: { author = {} },
+      },
+      actions: { updateStory },
+    }) => {
+      return {
+        isSaving,
+        author,
+        updateStory,
+      };
+    }
+  );
+
+  const [queriedUsers, setQueriedUsers] = useState(null);
+  const [visibleOptions, setVisibleOptions] = useState(null);
+
+  useEffect(() => {
+    if (tab === 'document') {
+      loadUsers();
+    }
+  }, [tab, loadUsers]);
+
+  const getAuthorsBySearch = useCallback(
+    (search) => {
+      return getAuthors(search).then((data) => {
+        const userData = data.map(({ id, name }) => ({
+          id,
+          name,
+        }));
+        setQueriedUsers(userData);
+      });
+    },
+    [getAuthors]
+  );
+
+  useEffect(() => {
+    if (users?.length) {
+      const currentAuthor = users.find(({ id }) => author.id === id);
+      if (!currentAuthor) {
+        setVisibleOptions([author, ...users]);
+      } else {
+        setVisibleOptions(users);
+      }
+    }
+  }, [author, users]);
+
+  const handleChangeAuthor = useCallback(
+    ({ id, name }) => {
+      updateStory({
+        properties: { author: { id, name } },
+      });
+    },
+    [updateStory]
+  );
+
+  const dropDownParams = {
+    hasSearch: true,
+    lightMode: true,
+    onChange: handleChangeAuthor,
+    getOptionsByQuery: getAuthorsBySearch,
+    selectedId: author.id,
+    dropDownLabel: __('Author', 'web-stories'),
+  };
+  return (
+    <Row>
+      {isUsersLoading || !visibleOptions ? (
+        <AdvancedDropDown
+          placeholder={__('Loading…', 'web-stories')}
+          disabled
+          primaryOptions={[]}
+          {...dropDownParams}
+        />
+      ) : (
+        <AdvancedDropDown
+          options={queriedUsers}
+          primaryOptions={visibleOptions}
+          searchResultsLabel={__('Search results', 'web-stories')}
+          disabled={isSaving}
+          aria-label={__('Author', 'web-stories')}
+          {...dropDownParams}
+        />
```

Comment:
Couldn't we collapse this into a single node? Just extract `isDisabled` and `placeholder` to `const`s above and we're more or less there, no?

Reply:
Didn't touch this part since it should be replaced anyway later with a "proper" dropdown, doesn't hurt to do it though.